app/controlplane/integrations/integrations_test.go

```diff
@@ -0,0 +1,167 @@
+//
+// Copyright 2023 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package integrations_test
+
+import (
+	"testing"
+
+	schemaapi "github.com/chainloop-dev/chainloop/app/controlplane/api/workflowcontract/v1"
+	"github.com/chainloop-dev/chainloop/app/controlplane/integrations"
+	"github.com/chainloop-dev/chainloop/app/controlplane/integrations/mocks"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewBaseIntegration(t *testing.T) {
```

Comment:
this test showcases different combinations of material and envelope types an integration can subscribe to.